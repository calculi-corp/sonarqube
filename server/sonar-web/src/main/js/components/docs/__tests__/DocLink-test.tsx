/*
 * SonarQube
 * Copyright (C) 2009-2022 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockAppState } from '../../../helpers/testMocks';
import { DocLink } from '../DocLink';

it('should render simple link', () => {
  expect(
    shallow(
      <DocLink appState={mockAppState({ canAdmin: false })} href="http://sample.com">
        link text
      </DocLink>
    )
  ).toMatchSnapshot();
});

it('should render documentation link', () => {
  expect(
    shallow(
      <DocLink appState={mockAppState({ canAdmin: false })} href="/foo/bar">
        link text
      </DocLink>
    )
  ).toMatchSnapshot();
});

it('should render sonarqube link on sonarqube', () => {
  const wrapper = shallow(
    <DocLink appState={mockAppState({ canAdmin: false })} href="/#sonarqube#/foo/bar">
      link text
    </DocLink>
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('SonarQubeLink').dive()).toMatchSnapshot();
});

it('should render sonarqube admin link on sonarqube for admin', () => {
  const wrapper = shallow(
    <DocLink appState={mockAppState({ canAdmin: true })} href="/#sonarqube-admin#/foo/bar">
      link text
    </DocLink>
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('SonarQubeAdminLink').dive()).toMatchSnapshot();
});

it('should not render sonarqube admin link on sonarqube for non-admin', () => {
  const wrapper = shallow(
    <DocLink appState={mockAppState({ canAdmin: false })} href="/#sonarqube-admin#/foo/bar">
      link text
    </DocLink>
  );
  expect(wrapper.find('SonarQubeAdminLink').dive()).toMatchSnapshot();
});

it('should render documentation anchor', () => {
  expect(
    shallow(
      <DocLink appState={mockAppState({ canAdmin: false })} href="#quality-profiles">
        link text
      </DocLink>
    )
  ).toMatchSnapshot();
});
