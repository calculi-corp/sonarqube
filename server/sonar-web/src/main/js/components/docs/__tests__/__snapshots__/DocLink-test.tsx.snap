// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render sonarqube admin link on sonarqube for non-admin 1`] = `
<Fragment>
  link text
</Fragment>
`;

exports[`should render documentation anchor 1`] = `
<a
  href="#"
  onClick={[Function]}
>
  link text
</a>
`;

exports[`should render documentation link 1`] = `
<Link
  to="/documentation/foo/bar"
>
  link text
</Link>
`;

exports[`should render simple link 1`] = `
<Fragment>
  <a
    href="http://sample.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    link text
  </a>
  <DetachIcon
    className="text-muted little-spacer-left little-spacer-right text-baseline"
    size={12}
  />
</Fragment>
`;

exports[`should render sonarqube admin link on sonarqube for admin 1`] = `
<SonarQubeAdminLink
  canAdmin={true}
  url="/#sonarqube-admin#/foo/bar"
>
  link text
</SonarQubeAdminLink>
`;

exports[`should render sonarqube admin link on sonarqube for admin 2`] = `
<Link
  target="_blank"
  to="/foo/bar"
>
  link text
</Link>
`;

exports[`should render sonarqube link on sonarqube 1`] = `
<SonarQubeLink
  url="/#sonarqube#/foo/bar"
>
  link text
</SonarQubeLink>
`;

exports[`should render sonarqube link on sonarqube 2`] = `
<Link
  target="_blank"
  to="/foo/bar"
>
  link text
</Link>
`;
